pathfinder/token_network.py

```diff
@@ -225,16 +226,21 @@ def weight(
                     0
                 )
 
-        for i in range(k):
+        for _ in range(MAX_DIJKSTRA_ITERATIONS):
```

Comment:
Why exactly 50? Can we somehow couple that to the number of requested paths?

Reply:
I just assumed the regular range of `k` would be within about 3 to 10. Debatable assumption of course.

I guess `max(MIN_PATH_REDUNDANCY, PATH_REDUNDANCY_FACTOR * k)` with `MIN_PATH_REDUNDANCY = 10` and `PATH_REDUNDANCY = 2` would work too. Thoughts?

Reply:
We should limit `k` anyways, but I'm not sure if 10 is low. What about a limit of 20?

The proposal sounds good to me. Maybe set `PATH_REDUNDANCY` a bit higher, right now it would have been 5.

Reply:
Amended, with 
```
MAX_PATHS_PER_REQUEST = 25
MIN_PATH_REDUNDANCY = 20
PATH_REDUNDANCY_FACTOR = 4
```
So, worst case 100 iterations to find 25 distinct paths.
Possible DoS case: 20 iterations to find 2 distinct paths.